File: Algo/python/ValidAnagram.py
''' https://leetcode.com/problems/valid-anagram/ '''

class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        d = {}
        if len(s) != len(t):
            return False
        for c in s:
            if c in d:
                d[c] = d[c] + 1
            else:
                d[c] = 1
        
        for c in t:
            if (c not in d) or (d[c] == 0):
                return False
            d[c] = d[c] - 1
        
        ##s = d.keys()
        ##for c in s:
        ##    if d[c] != 0:
        ##        return False
                
        return True

File: Algo/Java/ValidAnagram.java
/* https://leetcode.com/problems/valid-anagram/ */

/* to check if the given string is the valid anagram of other string*/
public class ValidAnagram
{
    public boolean isAnagram(String s, String t)
    {
        int[] array = new int[256];
        if(s==null || t==null)
            return false;
        if(s.length()!=t.length())
            return false;
        for(int i=0;i<s.length();i++)
        {
            array[s.charAt(i)]++;
            array[t.charAt(i)]--;
        }
        for(int i=0;i<array.length;i++)
        {
            if(array[i]!=0)
                return false;
        }
        return true;
    }

    public static void main(String args[])
    {
        ValidAnagram v = new ValidAnagram();
        String s = "rat";
        String t = "car";

        System.out.println(v.isAnagram(s,t));
    }
}
